odoo/addons/test_lint/tests/test_checkers.py

```diff
@@ -522,3 +522,66 @@ def test_gettext_repr(self):
         self.assertEqual(len(errors), 2)
         for error in errors:
             self.assertEqual(error["symbol"], "gettext-repr")
+
+    def test_missing_gettext(self):
+        """Test that missing gettext call raises 'missing-gettext'."""
+        node = """
+            from odoo.exceptions import UserError
+            def some_function():
+                raise UserError('This is not translated')
+            """
+
+        exit_code, errors = self.check(node)
+        self.assertNotEqual(exit_code, os.EX_OK)
+        self.assertEqual(len(errors), 1)
+        self.assertEqual(errors[0]["symbol"], "missing-gettext")
+
+    def test_missing_gettext_attr(self):
+        """
+        Test that missing gettext call raises 'missing-gettext'
+        inside a function of type `astroid.Attribute`.
+        """
+        node = """
+            from odoo import exceptions
+            def some_function():
+                raise exceptions.UserError('This is not translated')
+            """
+
+        exit_code, errors = self.check(node)
+        self.assertNotEqual(exit_code, os.EX_OK)
+        self.assertEqual(len(errors), 1)
+        self.assertEqual(errors[0]["symbol"], "missing-gettext")
+
+    def test_gettext_present(self):
+        """Test that no message is raised if gettext is present."""
+        node = """
+            from odoo.exceptions import UserError
+            def some_function():
+                raise UserError(_('This is translated'))
+            """
+
+        exit_code, errors = self.check(node)
+        self.assertEqual(exit_code, os.EX_OK)
+        self.assertEqual(len(errors), 0)
+
+    def test_check_accepted_arguments(self):
+        """Test that only specific argument types are accepted in UserError."""
+        node = """
+            from odoo.exceptions import UserError
+            def some_function():
+                some_var = 'This is not translated'
+                raise UserError(some_var)  # This should pass
+                raise UserError(_('This is translated'))  # This should pass
+                raise UserError('This is not translated')  # This should fail
+                raise UserError(12345)  # This should fail
```

Comment:
I'd make two separate tests, one for the patterns that are expected to pass, and one for the patterns that are not expected to pass